drivers/spi/spi_context.h

```diff
@@ -208,8 +208,10 @@ void spi_context_update_tx(struct spi_context *ctx, u8_t dfs, u32_t len)
 
 	ctx->tx_len -= len;
 	if (!ctx->tx_len) {
-		ctx->current_tx++;
-		ctx->tx_count--;
+		do {
+			ctx->current_tx++;
+			ctx->tx_count--;
+		} while (ctx->tx_count && ctx->current_tx->len == 0);
```

Comment:
what you are doing here does not exactly match your commit message. You'll need to change your commit message.

What is done here is avoiding to jump on a bogus buffer entry where pointer is set, but with a length of 0.
Fact that it affects spi peripherals is indirect.